File: Backend/Model/Student.cs
using Backend.Model.HelpingModels;
using Newtonsoft.Json;

namespace Backend.Model
{
    public class Student : SoftDeletable
    {
        public string FullName { get; set; }
        public string RollNumber { get; set; }
        public string Email { get; set; }
        public bool Gender { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string City { get; set; }

        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public Interest Interest { get; set; }

        public string Department { get; set; }
        public string DegreeTitle { get; set; }
        public string Subject { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }
}

File: frontend/src/types/Student.ts
import { Interest } from "./Interest";


export interface Student {
    id: number;
    fullName: string;
    rollNumber: string;
    email: string;
    gender: boolean;
    dateOfBirth: string;
    city: string;
    // interest: string;
    interest: Interest;
    department: string;
    degreeTitle: string;
    subject: string;
    startDate: string;
    endDate: string;
    isDeleted: boolean;
    createdBy: string;
    createdOn: Date;
    modifiedBy: string;
    modifiedOn: Date;
}


export interface Distributions {
    // "key": "string",
    // "value": "string"
    key: string
    value: string
}

export interface DailyStudentCreationData {
    // "date": "2023-12-11T00:00:00",
    // "studentCount": 2
    date: Date
    studentCount: number
}


export interface AddUpdateStudent {
    fullName: string;
    rollNumber: string;
    email: string;
    gender: number;
    dateOfBirth: string;
    city: string;
    interest: string;
    department: string;
    degreeTitle: string;
    subject: string;
    startDate: string;
    endDate: string;
}